src/router/handlers/resourceHandler.ts

```diff
@@ -41,8 +50,15 @@ export default class ResourceHandler implements CrudHandlerInterface {
     async create(resourceType: string, resource: any) {
         this.validator.validate(resourceType, resource);
 
-        const createResponse = await this.dataService.createResource({ resourceType, resource });
-        return createResponse.resource;
+        try {
+            const createResponse = await this.dataService.createResource({ resourceType, resource });
+            return createResponse.resource;
+        } catch (e) {
+            if (isInvalidResourceError(e)) {
+                throw new createError.BadRequest(e.message);
+            }
+            throw e;
+        }

```

Comment:
Isn't the error handler middleware already doing this?
https://github.com/awslabs/fhir-works-on-aws-routing/blob/mainline/src/router/routes/errorHandling.ts#L30

is this just to set a different error message? If so, It'd be better to remove the `Failed to parse request body as JSON` thing from the error handler middleware and move it to wherever that JSON validation error is being thrown. Then the error middleware can just use the incoming error message as is

Reply:
updated